packages/workspace/src/workspace/workspace.ts

```diff
@@ -251,21 +244,27 @@ export const loadWorkspace = async (config: ConfigSource, credentials: ConfigSou
       if (currentServices.includes(service)) {
         throw new ServiceDuplicationError(service)
       }
-      currentEnvConf().services = [...services(), service]
-      await config.set(WORKSPACE_CONFIG_NAME, workspaceConfigInstance(workspaceConfig))
+      workspaceConfig.envs = workspaceConfig.envs.map(e => {
+        if (e.name === currentEnv()) {
+          e.services = [...currentServices, service]
+        }
+        return e
+      })
```

Comment:
this is editing `workspaceConfig` in place anyway, might as well take advantage of it and reduce the amount of code
```suggestion
      currentEnvConf().services.push(service)
```
or as it was originally:
```suggestion
      currentEnvConf().services = [...currentServices, service]
```

Reply:
Changed it to how it was originally.
CurrentEnvConf.services -> Could be undefined, that's why it is option 2 and not 1.